frontend/pages/campaigns/[campaignId]/contribute.tsx

```diff
@@ -0,0 +1,169 @@
+import { Box, Typography, useMediaQuery } from '@mui/material';
+import { NextPage } from 'next';
+import Head from 'next/head';
+import { Container, Stack, useTheme } from '@mui/system';
+import CharitySelectionStep from '../../../components/redeem/steps/CharitySelectionStep';
+import useSWR from 'swr';
+import { Nullable } from '../../../types/utils';
+import { useRouter } from 'next/router';
+import api from '../../../frontendApis';
+import React, { useState } from 'react';
+import { Form, Formik, isInteger } from 'formik';
+import * as Yup from 'yup';
+import {
+  containerSx,
+  desktopFormContainerSx,
+  mobileFormContainerSx,
+} from '../../../styles/components/redeem/RedeemStyles';
+import PersonalContributionStep from '../../../components/redeem/steps/PersonalContributionStep';
+import VerifyStep from '../../../components/redeem/steps/VerifyStep';
+import RedeemLoading from '../../../components/redeem/RedeemLoading';
+import { messageContainerSx } from '../../../styles/campaigns/indexStyles';
+import { CampaignPublicData } from '../../../types/campaigns';
+import { SecondaryDonationFormData, SecondaryDonationPostData } from '../../../types/donations';
+import ContributeStepper from '../../../components/campaigns/contribute/ContributeStepper';
+
+const validationSchema = Yup.object().shape({
+  campaignCharityId: Yup.number().required('Campaign charity is required'),
+  amount: Yup.number()
+    .nullable()
```

Comment:
If they're contributing on their own accord, should we not allow $0? Otherwise it's pointless

Reply:
Added backend model validation too